modules/standard/raid/leader_controller.py

```diff
@@ -1,36 +1,81 @@
-from core.command_param_types import Const, Character
+from core.command_param_types import Const, Character, Options
 from core.db import DB
-from core.decorators import instance, command, timerevent, event
+from core.decorators import instance, command, timerevent, event, setting
+from core.setting_types import ColorSettingType, BooleanSettingType
 from core.text import Text
 from core.tyrbot import Tyrbot
+from core.access_service import AccessService
+from core.lookup.character_service import CharacterService
+from core.setting_service import SettingService
 from core.private_channel_service import PrivateChannelService
+from core.public_channel_service import PublicChannelService
 from modules.core.org_members.org_member_controller import OrgMemberController
 import time
 
 
 @instance()
 class LeaderController:
-    NOT_LEADER_MSG = "Error! You must be raid leader, or have higher access level than the raid leader to use this command."
+    NOT_LEADER_MSG = "Error! You must be raid leader, or have higher access " \
+                     "level than the raid leader to use this command."
 
     def __init__(self):
         self.leader = None
         self.last_activity = None
+        self.echo = False
 
     def inject(self, registry):
         self.db: DB = registry.get_instance("db")
         self.text: Text = registry.get_instance("text")
-        self.access_service = registry.get_instance("access_service")
-        self.character_service = registry.get_instance("character_service")
+        self.access_service: AccessService = registry.get_instance("access_service")
+        self.character_service: CharacterService = registry.get_instance("character_service")
         self.bot: Tyrbot = registry.get_instance("bot")
+        self.setting_service: SettingService = registry.get_instance("setting_service")
+
+    @setting(name="leader_echo_color", value="#00FF00",
+             description="Color with which the leader's messages will be echoed with")
+    def leader_echo_color(self):
+        return ColorSettingType()
+
+    @setting(name="leader_echo_to_org", value="True",
+             description="Echo leader's messages to org channel")
+    def leader_echo_to_org(self):
+        return BooleanSettingType()
+
+    @setting(name="leader_echo_to_priv", value="True",
+             description="Echo leader's messages to private channel")
+    def leader_echo_to_priv(self):
+        return BooleanSettingType()
```

Comment:
remove the `leader_echo_to_org` and `leader_echo_to_priv` settings and have it echo back to the channel it came from.  everything else looks good.